src/os/bluestore/BlueStore.cc

```diff
@@ -6521,16 +6537,30 @@ void BlueStore::_txc_finish_kv(TransContext *txc)
     finishers[n]->queue(txc->onreadable);
     txc->onreadable = NULL;
   }
-  if (txc->oncommit) {
-    finishers[n]->queue(txc->oncommit);
-    txc->oncommit = NULL;
-  }
-  while (!txc->oncommits.empty()) {
-    auto f = txc->oncommits.front();
-    finishers[n]->queue(f);
-    txc->oncommits.pop_front();
+  if (g_conf->bluestore_rtc_sync_completions && txc->kv_submitted_sync &&
+      txc->wal_txn && txc->osr->is_finisher_done()) {
```

Comment:
txc->wal_txn condition is needed..
It's hard to apply bluestore_rtc_sync_completions option to the transactions w/o data.
That's because some cases cannot pass FLAG_SYNC to the destructors calling context->complete()...
` ~RunOnDelete() { /* FLAG_SYNC can't be delivered to_run object... */`
`~FlushState() { /* FLAG_SYNC can't be delivered to this destructor... */`

So i added wal_txn condition(Transactions w/o no data) to exclude above cases